cd/scripts/release-controller.sh

```diff
@@ -119,24 +118,29 @@ pushd "$CODE_GEN_DIR" 1>/dev/null
   GOLANG_VERSION=${GOLANG_VERSION:-"$(go list -f \{\{.GoVersion\}\} -m)"}
 popd 1>/dev/null
 
+buildah manifest create $AWS_SERVICE
+
 # build controller image
-if ! buildah bud \
-  --quiet="$QUIET" \
-  -t "$AWS_SERVICE_DOCKER_IMG" \
-  -f "$CONTROLLER_IMAGE_DOCKERFILE_PATH" \
-  --build-arg service_alias="$AWS_SERVICE" \
-  --build-arg service_controller_git_version="$CHART_VERSION" \
-  --build-arg service_controller_git_commit="$SERVICE_CONTROLLER_GIT_COMMIT" \
-  --build-arg build_date="$BUILD_DATE" \
-  --build-arg golang_version="$GOLANG_VERSION" \
-  --build-arg go_arch="$GOARCH" \
-  "$DOCKER_BUILD_CONTEXT"; then
-  exit 2
-fi
+for GOARCH in amd64 arm64; do
+  if ! buildah bud \
+    --quiet="$QUIET" \
+    -t "$AWS_SERVICE_DOCKER_IMG" \
+    -f "$CONTROLLER_IMAGE_DOCKERFILE_PATH" \
+    --build-arg service_alias="$AWS_SERVICE" \
+    --build-arg service_controller_git_version="$CHART_VERSION" \
+    --build-arg service_controller_git_commit="$SERVICE_CONTROLLER_GIT_COMMIT" \
+    --build-arg build_date="$BUILD_DATE" \
+    --build-arg golang_version="$GOLANG_VERSION" \
+    --build-arg go_arch="$GOARCH" \
+    --arch $GOARCH \
+    "$DOCKER_BUILD_CONTEXT"; then
+    exit 2
+  fi
+done
 
 echo "Pushing '$AWS_SERVICE' controller image with tag: ${AWS_SERVICE_DOCKER_IMG}"
 
-if ! buildah push "${AWS_SERVICE_DOCKER_IMG}"; then
+if ! buildah manifest push --all "${$AWS_SERVICE}" "${AWS_SERVICE_DOCKER_IMG}"; then
```

Comment:
i would like to test this locally and see how it impacts the helm chart. Also i want to know how does a cluster pull the right image, or does it pull all the images?

Reply:
The k8s cluster downloads only one container image from the manifest list, i.e., the one that matches its environment.

Reply:
Reference information on manifestlist if you like.
https://docs.docker.com/engine/reference/commandline/manifest/